composer/loggers/progress_bar_logger.py

```diff
@@ -229,62 +225,67 @@ def _build_pbar(self, state: State, is_train: bool, epoch_style: bool = False) -
             Epoch     0 train 100%|█████████████████████████| 29/29
             Epoch     1 train 100%|█████████████████████████| 29/29
         """
-        position = 0 if is_train else 1
-        split = 'train' if is_train else 'val'
+        position = 0
+        if self.dummy_pbar is not None:
+            position += 1
+        if self.train_pbar is not None:
+            position += 1
+        label = state.dataloader_label
+        assert state.max_duration is not None, 'max_duration should be set'
 
-        assert self.is_train is not None
         if epoch_style:
             total = int(state.dataloader_len) if state.dataloader_len is not None else None
 
             # handle when # batches is less than an epoch
-            if total is not None and state.max_duration is not None \
-                and state.max_duration.unit == TimeUnit.BATCH:
+            if total is not None and state.max_duration.unit == TimeUnit.BATCH:
                 total = min(total, state.max_duration.value)
 
             unit = TimeUnit.BATCH
             n = state.timestamp.epoch.value
-            if not is_train:
-                # eval results refer to model from previous epoch (n-1)
-                n = max(0, n - 1)
-            desc = f'Epoch {n:5d} {split:5s}'
+            if self.train_pbar is None and not is_train:
+                # epochwise eval results refer to model from previous epoch (n-1)
+                n -= 1
```

Comment:
The `max` was needed before because mid-epoch evaluation progress bars followed this codepath. With the refactor, `self.train_pbar` would be not None if evaluating mid-epoch, causing this line to be skipped. 

Verified this works manually:

```yaml
train_dataset:
  &train_dataset
  mnist:
    use_synthetic: true
evaluators:
  eval1:
    label: eval/batch
    eval_interval: 1ba
    subset_num_batches: 2
    eval_dataset:
      <<: *train_dataset
model:
  mnist_classifier:
    num_classes: 10
max_duration: 1ep
train_batch_size: 2000
eval_batch_size: 2000
dataloader:
  pin_memory: true
  timeout: 0
  prefetch_factor: 2
  persistent_workers: true
  num_workers: 1
train_subset_num_batches: 2
```

Output:
```
Epoch   0, Batch   1: eval/batch      100%|█████████████████████████| 2/2 [00:00<00:00,  5.45ba/s]                                                                                                   
Epoch   0, Batch   2: eval/batch      100%|█████████████████████████| 2/2 [00:00<00:00,  5.22ba/s]                                                                                                   
Epoch   0:            train           100%|█████████████████████████| 2/2 [00:01<00:00,  1.38ba/s, loss/train=2.2977] 
```